backends/graphs/graphs.cpp

```diff
@@ -0,0 +1,128 @@
+/*
+Copyright 2013-present Barefoot Networks, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include "ir/ir.h"
+#include "ir/json_loader.h"
+#include "lib/log.h"
+#include "lib/error.h"
+#include "lib/exceptions.h"
+#include "lib/gc.h"
+#include "lib/crash.h"
+#include "lib/nullstream.h"
+#include "frontends/common/parseInput.h"
+#include "frontends/p4/evaluator/evaluator.h"
+#include "frontends/p4/frontend.h"
+
+#include "config.h"
+#ifdef HAVE_LIBBOOST_GRAPH
+#include "controls.h"
+#endif
+
+namespace graphs {
+
+class MidEnd : public PassManager {
+ public:
+    P4::ReferenceMap    refMap;
+    P4::TypeMap         typeMap;
+    IR::ToplevelBlock   *toplevel = nullptr;
+
+    explicit MidEnd(CompilerOptions& options);
+    IR::ToplevelBlock* process(const IR::P4Program *&program) {
+        program = program->apply(*this);
+        return toplevel;
+    }
+};
+
+MidEnd::MidEnd(CompilerOptions& options) {
+    bool isv1 = options.langVersion == CompilerOptions::FrontendVersion::P4_14;
+    refMap.setIsV1(isv1);
+    auto evaluator = new P4::EvaluatorPass(&refMap, &typeMap);
+    setName("MidEnd");
+
+    addPasses({
+        evaluator,
+        new VisitFunctor([this, evaluator]() { toplevel = evaluator->getToplevelBlock(); }),
+    });
+}
+
+class Options : public CompilerOptions {
+ public:
+    cstring graphsDir{"."};
+    Options() {
+        registerOption("--graphs-dir", "dir",
```

Comment:
You could also just reuse the existing -o option and treat it as a folder name, or as a filename prefix.
But there is no simple way to change its documentatition...